addons/event/models/event_event.py

```diff
@@ -208,7 +206,7 @@ def _compute_seats(self):
 
         # compute seats_available
         for event in self:
-            event.update(results[event._origin.id or event.id])
+            event.update(results.get(event._origin.id or event.id, base_vals))
```

Comment:
Not sure how `event._origin.id` could be in your dict when you're using self.ids to initialize and populate it.

Or by some black magic self.ids would actually contain `event._origin.id` ?

Reply:
it should yes.
If no _origin.id, that means it's a new record and if new record, event.id will be the fallback. 
A priori, origin.id and event.id are always equal once origin.id is set.
We take the origin.id by default to avoid newId<xxx> errors. Am I right ?

Reply:
Alors after regarding un peu.

Kanban computation -> self.ids and self._origin.ids contain real ids
Onchange -> self.ids and self._origin.ids contain record ids, but record.id is a newId where record._origin.id is a real id
Create -> self.ids and self._origin.ids are void list, record.id is a newId and record._origin.id is False

So, doing event._origin.id ensure that in onchange mode we have a real id. And the result dict is build upon self.ids that is a real list of ids (or void in create but that's ok for us).